classes/manager.php

```diff
@@ -792,6 +792,7 @@ public static function possible_factor_setup() {
         // Check if there are enough factors that a user can ONLY pass, if so, don't display the menu.
         $weight = 0;
         foreach ($factors as $factor) {
+            if ( $factor->name == "nosetup") { continue; }
```

Comment:
Im not sure this is the best way to approach this, it feels the possible_states in the factor for this. Eg it should return a STATE_NEUTRAL always instead of a STATE_PASS, with a decent explainer comment that to support optional rollouts correctly, the state needs to be exposed as NEUTRAL otherwise the menu will not display.